nixos/modules/services/networking/xl2tpd.nix

```diff
@@ -34,6 +34,31 @@ with lib;
           ms-dns 8.8.4.4
         '';
       };
+
+      up = mkOption {
+        type        = types.lines;
+        description = "Contents of up script.";
+        default     = "";
+        example     = literalExample ''
+          ''${pkgs.nettools}/bin/route add -net 192.168.1.0 netmask 255.255.255.0 gw 192.168.100.1 dev ppp0
+          ''${pkgs.coreutils}/bin/cat /etc/xl2tpd/ppp/resolv.conf | ''${pkgs.openresolv}/sbin/resolvconf -m 0 -a ppp0
+        '';
+      };
+
+      down = mkOption {
+        type        = types.lines;
+        description = "Contents of down script.";
+        default     = "";
+        example     = literalExample ''
+          ''${pkgs.openresolv}/sbin/resolvconf -d ppp0
```

Comment:
To demonstrate that nixos isn't magic, it might be good to have `route add`'s `down` counterpart.